packages/app/src/vis-packs/nexus/NxValuesFetcher.tsx

```diff
@@ -15,36 +16,32 @@ interface Props<T extends NumericType | ComplexType> {
 
 function NxValuesFetcher<T extends NumericType | ComplexType>(props: Props<T>) {
   const { nxData, selection, render } = props;
-  const {
-    signalDataset,
-    errorDataset,
-    axisDatasets,
-    auxDatasets,
-    titleDataset,
-  } = nxData;
+  const { signalDef, axisDefs, auxDefs, titleDataset } = nxData;
 
+  const axisDatasets = axisDefs.map((def) => def?.dataset);
+  const auxDatasets = auxDefs.map((def) => def?.dataset);
+  const auxErrorDatasets = auxDefs.map((def) => def?.errorDataset);
+
+  usePrefetchValues([titleDataset, ...axisDatasets]);
   usePrefetchValues(
     [
-      signalDataset,
-      errorDataset,
-      ...auxDatasets.flatMap((d) => [d.signal, d.errors]),
+      signalDef.dataset,
+      signalDef.errorDataset,
+      ...auxDatasets,
+      ...auxErrorDatasets,
     ],
     selection
   );
-  usePrefetchValues([...axisDatasets, titleDataset]);
 
-  const { attrValuesStore } = useDataContext();
-  const signal = useDatasetValue(signalDataset, selection);
-  const signalLabel = getDatasetLabel(signalDataset, attrValuesStore);
-  const errors = useDatasetValue(errorDataset, selection);
-  const axisValues = useAxisValues(axisDatasets);
-  const auxiliaries = useAuxiliaries(auxDatasets, selection);
-  const title = useDatasetValue(titleDataset) || signalLabel;
+  const title = useDatasetValue(titleDataset) || signalDef.label;
+  const signal = useDatasetValue(signalDef.dataset, selection);
+  const errors = useDatasetValue(signalDef.errorDataset, selection);
+  const auxValues = useDatasetValues(auxDatasets, selection);
+  const auxErrors = useDatasetValues(auxErrorDatasets, selection);
+  const axisValues = useDatasetValues(axisDatasets);
 
   return (
-    <>
-      {render({ signal, signalLabel, errors, axisValues, auxiliaries, title })}
-    </>
+    <>{render({ title, signal, errors, auxValues, auxErrors, axisValues })}</>
```

Comment:
Is there a reason for the change in order ? I liked the fact that the variables were sorted by importance

Reply:
I thought this order was conveying importance better 😂 :

1. title of the plot because it tells you what you're looking at ;
2. signal and errors because that's the important plot
3. aux and aux errors because they're still data being plotted, so quite important
4. axis values because I felt they were less important than the rest

Axis stuff between signal and auxiliaries didn't feel right. I don't mind moving the `title` before `axisValues`, since it's a kind of metadata, but as I said, I think the title is actually the most important info of a NeXus plot.

Reply:
Yeah but `title` can be `undefined`, no ?

Anyway, do not let this stop you from merging, it is fine.

Reply:
Yeah you're right, I can move it down 👍

Reply:
Actually no, it's always defined since there's a fallback on `signalDef.label`. I'll leave it at the top.